mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergFilterFactory.java

```diff
@@ -0,0 +1,172 @@
+    switch (leaf.getOperator()) {
+      case EQUALS:
+        return equal(column, leafToLiteral(leaf));
+      case LESS_THAN:
+        return lessThan(column, leafToLiteral(leaf));
+      case LESS_THAN_EQUALS:
+        return lessThanOrEqual(column, leafToLiteral(leaf));
+      case IN:
+        return in(column, leafToLiteralList(leaf));
+      case BETWEEN:
+        List<Object> icebergLiterals = leafToLiteralList(leaf);
+        return and(greaterThanOrEqual(column, icebergLiterals.get(0)),
+                lessThanOrEqual(column, icebergLiterals.get(1)));
+      case IS_NULL:
+        return isNull(column);
+      default:
+        throw new UnsupportedOperationException("Unknown operator: " + leaf.getOperator());
+    }
+  }
+
+  private static Object leafToLiteral(PredicateLeaf leaf) {
+    switch (leaf.getType()) {
+      case LONG:
+      case BOOLEAN:
+      case STRING:
+      case FLOAT:
+        return leaf.getLiteral();
+      case DATE:
+        //Hive converts a Date type to a Timestamp internally when retrieving literal
+        return timestampToDateString((Timestamp) leaf.getLiteral());
+      case DECIMAL:
+        return hiveDecimalToBigDecimal((HiveDecimalWritable) leaf.getLiteral());
+      case TIMESTAMP:
+        return timestampToUnixEpoch((Timestamp) leaf.getLiteral());
+
+      default:
+        throw new UnsupportedOperationException("Unknown type: " + leaf.getType());
+    }
+  }
+
+  private static List<Object> leafToLiteralList(PredicateLeaf leaf) {
+    switch (leaf.getType()) {
+      case LONG:
+      case BOOLEAN:
+      case FLOAT:
+      case STRING:
+        return leaf.getLiteralList();
+      case DATE:
+        return leaf.getLiteralList().stream().map(value -> dateToString((Date) value))
+                .collect(Collectors.toList());
+      case DECIMAL:
+        return leaf.getLiteralList().stream()
+                .map(value -> hiveDecimalToBigDecimal((HiveDecimalWritable) value))
+                .collect(Collectors.toList());
+      case TIMESTAMP:
+        return leaf.getLiteralList().stream()
+                .map(value -> timestampToUnixEpoch((Timestamp) value))
+                .collect(Collectors.toList());
+      default:
+        throw new UnsupportedOperationException("Unknown type: " + leaf.getType());
+    }
+  }
+
+  private static String timestampToDateString(Timestamp timestamp) {
+    return timestamp.toLocalDateTime().toLocalDate().toString();
+  }
+
+  private static String dateToString(Date date) {
+    return date.toLocalDate().toString();
+  }
+
+  private static BigDecimal hiveDecimalToBigDecimal(HiveDecimalWritable hiveDecimalWritable) {
+    return hiveDecimalWritable.getHiveDecimal().bigDecimalValue().setScale(hiveDecimalWritable.scale());
+  }
+
+  private static long timestampToUnixEpoch(Timestamp timestamp) {
+    return timestamp.toInstant().getEpochSecond() * TimeUnit.SECONDS.toMicros(1) +
+            timestamp.getNanos() / TimeUnit.MICROSECONDS.toNanos(1);
```

Comment:
This seems odd to me. Why not call `TimeUnit.SECONDS.toMicros(timestamp.toInstant().getEpochSecond())`? Using the `toMicros` function to get the conversion factor, but not actually using it for conversion is strange.

Reply:
+1, similarly for `timestamp.getNanos() / TimeUnit.MICROSECONDS.toNanos(1)` -`TimeUnit.NANOSECONDS.toMicros(timestamp.getNanos())`

Reply:
Ah yep, probably should have spotted that one 😅